### PR TITLE
feat: Add Loading component and use it in SearchInput

### PR DESCRIPTION
**Pull request by luis-herasme**
### Checklist

- [ ] My change requires a documentation update, and I have done it.
- [ ] I have added tests to cover my changes.
- [x] I have filled out the description and linked the related issues.

### Description

A `Loading` component is added to provide a loading animation while fetching search results in the `SearchInput` component. This improves the user experience by indicating that the search is in progress.
### Related Issue

#289

### Screenshots:
image

### CODE REVIEW

File: apps/web/src/components/Loading.tsx
Comment (PJColombo) [inline apps/web/src/components/Loading.tsx:1]: There's no need to create a component wrapping a svg. We configured [`svgr`](https://react-svgr.com) for that which does it automatically.

Let's add this to an svg file to [`/src/icons`](https://github.com/Blobscan/blobscan/tree/next/apps/web/src/icons) which is the folder that `svgo` uses to optimise svgs images

**Review by PJColombo — CHANGES_REQUESTED** [review]
Nice job!.

Left a comment.

Can we also display a "No results found" message in the search results modal when no results are returned? Currently, the loading spinner appears and disappears too quickly, which might confuse users as there is no indication whether results were found.

File: apps/web/src/components/Loading.tsx
Comment (PJColombo) [reply apps/web/src/components/Loading.tsx:1]: Also. Can you run `pnpm svg:format` from the `web` app package to optimise the images once you add them?.  

We need to add this to the pipeline to do it automatically

**Comment by luis-herasme** [comment]
> Currently, the loading spinner appears and disappears too quickly,

I improved this after the last commit. Now, we show the loading spinner while debouncing.

**Comment by luis-herasme** [comment]
> Currently, the loading spinner appears and disappears too quickly,

I improved this after the last commit. Now, we show the loading spinner while debouncing.

**Comment by luis-herasme** [comment]
No results found component:
image

**Review by PJColombo — APPROVED** [review]
LGTM!

One last thing: let's add a changeset

**Pull request merged**